src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAroundBasic.java

```diff
@@ -1,277 +1,247 @@
+    int b= 1; // violation ''=' is not preceded with whitespace.'
+    b= 1; // violation ''=' is not preceded with whitespace.'
+    b +=1; // violation ''\+=' is not followed by whitespace.'
+    b -=- 1 + (+ b); // violation ''-=' is not followed by whitespace.'
+    b = b++ + b--; // ok
+    b = ++ b - -- b; // ok
+  }
+
+  /** method * */
+  void method2() {
+    synchronized(this) {
+      // 2 violations above:
+      //  ''synchronized' is not followed by whitespace.'
+      //  ''synchronized' is not followed by whitespace.'
+    }
+    try{
+      // 3 violations above:
+      //  ''try' is not followed by whitespace.'
+      //  ''try' is not followed by whitespace.'
+      //  ''{' is not preceded with whitespace.'
+    } catch (RuntimeException e) {// violation ''{' is not followed by whitespace.'
+    }
+  }
+
+  /** test WS after void return */
+  private void fastExit() {
+    boolean complicatedStuffNeeded = true;
+    if(!complicatedStuffNeeded)
+    // 2 violations above:
+    //  ''if' is not followed by whitespace.'
+    //  ''if' is not followed by whitespace.'
+    {
+      // should not complain about missing WS after return
+    } else {
+      // do complicated stuff
+    }
+  }
+
+  /**
+   * test WS after non void return
+   *
+   * @return 2
+   */
+  private int nonVoid() {
+    if (true) {
+      return(2);
+      // 2 violations above:
+      //  ''return' is not followed by whitespace.'
+      //  ''return' is not followed by whitespace.'
+    } else {
+      return 2; // this is ok
+    }
+  }
+
+  /** test casts * */
+  private void testCasts() {
+    Object o = new Object(); // ok
+    o = o; // ok
+    o = o; // ok
+    o = o; // ok
+  }
+
+  /** test questions * */
+  private void testQuestions() {
+
+    boolean b = (1 ==2) ? false : true; // violation ''==' is not followed by whitespace.'
+  }
+
+  /** star test * */
+  private void starTest() {
+    int x = 2 * 3* 4; // violation ''*' is not preceded with whitespace.'
+  }
+
+  /** boolean test * */
+  private void boolTest() {
+    boolean a = true;
+    boolean x = !a;
+    int z = ~1 + ~2;
+  }
+
+  /** division test * */
+  private void divTest() {
+    int a = 4 % 2;
+    int b = 4% 2; // violation ''%' is not preceded with whitespace.'
+    int c = 4 %2; // violation ''%' is not followed by whitespace.'
+    int d = 4% 2; // violation ''%' is not preceded with whitespace.'
+    int e = 4 / 2;
+    int f = 4/ 2; // violation ''/' is not preceded with whitespace.'
+    int g = 4 /2; // violation ''/' is not followed by whitespace.'
+  }
+
+  /**
+   * @return dot test *
+   */
+  private java.lang.String dotTest() {
+    Object o = new java.lang.Object();
+    o.toString();
+    o.toString();
+    o.toString();
+    return o.toString();
+  }
+
+  /** assert statement test */
+  public void assertTest() {
+    // OK
+    assert true;
+
+    // OK
+    assert true : "Whups";
+
+    // evil colons, should be OK
+    assert !"OK".equals(null) : "Whups";
```

Comment:
Was it done by a formatter?

Reply:
yes, original code was:

```java
        // evil colons, should be OK
        assert "OK".equals(null) ? false : true : "Whups";
```

Reply:
I guess formatter have made these types of modifications in few other places also.

Reply:
Interesting, it's no longer a formatter if it does such things :)

Reply:
Yeah I was also surprised by this 😅

Do I need to revert this change?

Reply:
I tried formatting the `assert "OK".equals(null) ? false : true : "Whups";` in VS code with google-java-format plugin and it didn't changed anything but in IntelliJ Idea when I formatted the code using the same plugin, it formatted it to `assert !"OK".equals(null) : "Whups";`.

Don't know if the formatter has some bug or is it because of the IDE.

Reply:
Omg, yes, it is not a formatter.

Fortunately in this code we don't much care on content of this code.